clang/test/CIR/Lowering/region-simplify.cir

```diff
@@ -1,5 +1,5 @@
-// RUN: cir-opt %s -canonicalize -cir-to-llvm -o - | FileCheck %s -check-prefix=MLIR
-// RUN: cir-opt %s -canonicalize -o - | cir-translate -cir-to-llvmir  | FileCheck %s -check-prefix=LLVM
+// RUN: cir-opt %s -cir-to-llvm -canonicalize -o - | FileCheck %s -check-prefix=MLIR
```

Comment:
Not sure I understand this change, does this make the canonicalizer run after we lower down to LLVM?

Reply:
Yes. I want to preserve the effect presented by the original test file as much as possible, so I added cir.goto in this test file to prevent ^bb1 from being folded. Because cir.goto was added, we need to correctly handle goto in cir-to-llvm first. If we run -canonicalize first, it will cause cir.label to be removed, resulting in the following error:
./clang/test/CIR/Lowering/region-simplify.cir:7:3: error: 'cir.func' op goto/label mismatch